portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -98,8 +103,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       Gson gson = new Gson();
       response.setContentType("application/json;");
       
-      response.getWriter().println(gson.toJson(loadedBlogMessages));
-      return;
+      if(numberOfCommentsToDisplay == 0){
+        response.getWriter().println(gson.toJson(allBlogMessages)); // set a default amount later.
+        return;
+      } else {
+        List<BlogMessage> limitedBlogMessages = new ArrayList<>();
+        for (int i = 0; i < allBlogMessages.size(); i++) {
```

Comment:
This seems to copy allBlogMessages into limitedBlogMessages. I think you meant to use numberOfCommentsToDisplay?